modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbar.ts

```diff
@@ -0,0 +1,276 @@
+    ));
+
+    return renderToolbar({
+      type: toolbarType,
+      uid: Id.generate('context-toolbar'),
+      initGroups,
+      onEscape: Optional.none,
+      cyclicKeying: true,
+      providers: sharedBackstage.providers
+    });
+  };
+
+  const getAnchor = (position: InlineContent.ContextPosition, element: Optional<SugarElement<Element>>): AnchorSpec => {
+    const anchorage = position === 'node' ? sharedBackstage.anchors.node(element) : sharedBackstage.anchors.cursor();
+    const anchorLayout = ContextToolbarAnchor.getAnchorLayout(editor, position, isTouch(), {
+      lastPos: lastPosition.get,
+      lastElement: lastElement.get,
+      isReposition: () => Optionals.is(lastTrigger.get(), TriggerCause.Reposition),
+      bounds: lastBounds.get
+    });
+    return Merger.deepMerge(anchorage, anchorLayout);
+  };
+
+  const launchContext = (toolbarApi: Array<ContextType>, elem: Optional<Element>) => {
+    launchContextToolbar.stop();
+
+    // If a mobile context menu is open, don't launch else they'll probably overlap. For android, specifically.
+    if (isTouch() && backstage.isContextMenuOpen()) {
+      return;
+    }
+
+    const toolbarSpec = buildToolbar(toolbarApi);
+    const sElem = elem.map(SugarElement.fromDom);
+
+    // TINY-4495 ASSUMPTION: Can only do toolbarApi[0].position because ContextToolbarLookup.filterToolbarsByPosition
+    // ensures all toolbars returned by ContextToolbarLookup have the same position.
+    // And everything else that gets toolbars from elsewhere only returns maximum 1 toolbar
+    const anchor = getAnchor(toolbarApi[0].position, sElem);
+    lastTrigger.set(TriggerCause.NewAnchor);
+
+    const contextBarEle = contextbar.element;
+    Css.remove(contextBarEle, 'display');
+    InlineView.showWithinBounds(contextbar, anchor, wrapInPopDialog(toolbarSpec), () => Optional.some(getBounds()));
+
+    // IMPORTANT: This must be stored after the initial render, otherwise the lookup of the last element in the
+    // anchor placement will be incorrect as it'll reuse the new element as the anchor point.
+    sElem.fold(lastElement.clear, lastElement.set);
+    lastPosition.set(Boxes.box(contextbar.element));
+
+    // It's possible we may have launched offscreen, if so then hide
+    if (shouldContextToolbarHide()) {
+      Css.set(contextBarEle, 'display', 'none');
+    }
+  };
+
+  const launchContextToolbar = Delay.debounce(() => {
+    // Don't launch if the editor doesn't have focus or has been destroyed
+    if (!editor.hasFocus() || editor.removed) {
+      return;
+    }
+
+    const scopes = getScopes();
+    ToolbarLookup.lookup(scopes, editor).fold(
+      close,
+
+      (info) => {
+        launchContext(info.toolbars, Optional.some(info.elem.dom));
+      }
+    );
+  }, 0);
+
+  editor.on('init', () => {
+    editor.on('remove', close);
+    editor.on('ScrollContent ScrollWindow ObjectResized ResizeEditor longpress', hideOrRepositionIfNecessary);
```

Comment:
`ObjectResized` and `ResizeEditor` were intentionally moved here, as they were added to the `launchContextToolbar` before the `hideOrRepositionIfNecessary` function existed. As neither of these events can change the selection/content they aren't going to be able to launch anything and as such only need to handle repositioning instead.